core/lib/state/src/catchup.rs

```diff
@@ -174,15 +188,60 @@ impl AsyncCatchupTask {
             .await
             .context("Failed to catch up RocksDB to Postgres")?;
         drop(connection);
+
         if let Some(rocksdb) = rocksdb {
             self.db_sender.send_replace(Some(rocksdb.into_rocksdb()));
         } else {
             tracing::info!("Synchronizing RocksDB interrupted");
         }
+
         Ok(())
     }
 }
 
+#[derive(Debug)]
+pub struct KeepUpdatedTask {
+    db: AsyncOnceCell<RocksDB<StateKeeperColumnFamily>>,
+    pool: ConnectionPool<Core>,
+}
+
+impl KeepUpdatedTask {
+    pub async fn run(mut self, mut stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        const SLEEP_INTERVAL: Duration = Duration::from_millis(100);
+
+        let db: RocksDB<StateKeeperColumnFamily> = tokio::select! {
+            res = self.db.wait_for(Option::is_some) => match res {
+                // `unwrap()` is safe by construction
+                Ok(db) => db.clone().unwrap(),
+                // Interrupted
```

Comment:
Nit: May make sense to log interruption here and below (in another `select!` clause and when processing `RocksdbSyncError::Interrupted`).